int8 id
float32 orientation